state-sync/src/request_manager.rs

```diff
@@ -137,8 +137,11 @@ impl RequestManager {
         if let Some(peer_info) = self.peers.get_mut(&peer) {
             peer_info.is_alive = true;
         } else {
-            self.peers.insert(peer, PeerInfo::new(true, MAX_SCORE));
+            self.peers
+                .insert(peer.clone(), PeerInfo::new(true, MAX_SCORE));
         }
+
+        self.lower_multicast_level_to_match_peer_level(&peer);
```

Comment:
I thought this is intentional not a bug? a connected validator may not send back a valid response (imagine it crashed and is still catching up)

lower the multicast level too early will make the node failover again

Reply:
(FYI: this might be me misunderstanding how new and lost peer events happen)

- I understand that lowering the multicast level too early might result in us falling back (in the case the connecting validator crashed and is now back online).
- However, are there not other possible cases here? e.g., what about the case where *we* see some network issues (and for whatever reason can't send/receive network packets), so we start falling back. The online validators might then disconnection from us (e.g., a timeout, because we're not responding to anything).. and then when we come back online (the network issue is resolved), they might send a new peer event to us, in which case we want to lower the multicast level (because we're now connected to up-to-date validators)?


Overall though, this is a pretty specific/tiny edge-case that we shouldn't see too frequently. If you'd prefer I update the test so that we don't automatically lower the multicast level, let me know. I don't feel too strongly here given that this won't happen frequently 😄

Reply:
the only reason I remember this is the first case happened before and we need to fix it.
for the first case the fullnode would need to timeout again before it can failover to others, for the second case, it's just slower to reconnect to validators but will continue to get data from failovers so I'd slightly prefer the current approach

Reply:
@zekun000, okay, thanks! Makes sense to me. I've updated the PR to just fix and clean up the tests. Note: it seems like I've managed to fix both tests -- i.e., I don't get any flaky behaviours locally. Hopefully it stays that way going forward, but if we land this, I'll keep an eye out.